File: tests/expected/infer_ops.cpp
#include <cassert>   // NOLINT(build/include_order)
#include <iostream>  // NOLINT(build/include_order)

#include "pycpp/runtime/builtins.h"  // NOLINT(build/include_order)
#include "pycpp/runtime/sys.h"       // NOLINT(build/include_order)

inline void foo() {
  int a = 10;
  int b = 20;
  int c1 = a + b;
  assert(c1 == 30);
  int c2 = a - b;
  assert(c2 == -10);
  int c3 = a * b;
  assert(c3 == 200);
  double c4 = a / b;
  assert(c4 == 0.5);
  int c5 = -(a);
  assert(c5 == -10);
  double d = 2.0;
  double e1 = a + d;
  assert(e1 == 12.0);
  double e2 = a - d;
  assert(e2 == 8.0);
  double e3 = a * d;
  assert(e3 == 20.0);
  double e4 = a / d;
  assert(e4 == 5.0);
  double f = -3.0;
  assert(f < -2.9);
}

inline int16_t add1(int8_t x, int8_t y) { return x + y; }

inline int32_t add2(int16_t x, int16_t y) { return x + y; }

inline int64_t add3(int32_t x, int32_t y) { return x + y; }

inline int64_t add4(int64_t x, int64_t y) { return x + y; }

inline uint16_t add5(uint8_t x, uint8_t y) { return x + y; }

inline uint32_t add6(uint16_t x, uint16_t y) { return x + y; }

inline uint64_t add7(uint32_t x, uint32_t y) { return x + y; }

inline uint64_t add8(uint64_t x, uint64_t y) { return x + y; }

inline uint32_t add9(int8_t x, uint16_t y) { return x + y; }

inline int8_t sub(int8_t x, int8_t y) { return x - y; }

inline int16_t mul(int8_t x, int8_t y) { return x * y; }

inline double fadd1(int8_t x, double y) { return x + y; }

inline void show() {
  foo();
  assert(fadd1(6, 6.0) == 12);
  std::cout << std::string{"OK"};
  std::cout << std::endl;
}

int main(int argc, char** argv) {
  pycpp::sys::argv = std::vector<std::string>(argv, argv + argc);
  foo();
  show();
}

File: tests/expected/infer_ops.dart
// @dart=2.9
import 'package:sprintf/sprintf.dart';

foo() {
  final int a = 10;
  final int b = 20;
  final int c1 = (a + b);
  assert(c1 == 30);
  final int c2 = (a - b);
  assert(c2 == -10);
  final int c3 = (a * b);
  assert(c3 == 200);
  final int c4 = (a ~/ b);
  assert(c4 == 0.5);
  final int c5 = -(a);
  assert(c5 == -10);
  final double d = 2.0;
  final double e1 = (a + d);
  assert(e1 == 12.0);
  final double e2 = (a - d);
  assert(e2 == 8.0);
  final double e3 = (a * d);
  assert(e3 == 20.0);
  final double e4 = (a / d);
  assert(e4 == 5.0);
  final double f = -3.0;
  assert(f < -2.9);
}

int add1(int x, int y) {
  return (x + y);
}

int add2(int x, int y) {
  return (x + y);
}

int add3(int x, int y) {
  return (x + y);
}

int add4(int x, int y) {
  return (x + y);
}

int add5(int x, int y) {
  return (x + y);
}

int add6(int x, int y) {
  return (x + y);
}

int add7(int x, int y) {
  return (x + y);
}

int add8(int x, int y) {
  return (x + y);
}

int add9(int x, int y) {
  return (x + y);
}

int sub(int x, int y) {
  return (x - y);
}

int mul(int x, int y) {
  return (x * y);
}

double fadd1(int x, double y) {
  return (x + y);
}

show() {
  foo();
  assert(fadd1(6, 6.0) == 12);
  print(sprintf("%s", ["OK"]));
}

main(List<String> argv) {
  foo();
  show();
}
